### PR TITLE
multiregion: job stop across regions with -global flag

### PR DESCRIPTION
**Pull request by tgross**
Fixes https://github.com/hashicorp/nomad/issues/8198

Adds a `-global` flag for stopping multiregion jobs in all regions at once. Warn the user if they attempt to stop a multiregion job in a single region.

---

In use:

```
$ nomad job run multiregion.nomad
Job registration successful
Evaluation ID: f13081e1-5a49-b98c-f49b-c1a1245498d5

$ nomad job status example
...
Status        = running
...
Allocations
ID        Node ID   Task Group  Version  Desired  Status    Created    Modified
204c970f  4786abea  cache       0        run      running   8s ago     7s ago

$ nomad job stop -global example
16e1bccc-c26c-32e1-92ca-71b502e3ebf6

$ nomad job status example
...
Status        = dead (stopped)
...
Allocations
ID        Node ID   Task Group  Version  Desired  Status    Created    Modified
204c970f  4786abea  cache       0        stop     complete  24s ago    4s ago

$ nomad job run multi.nomad
Job registration successful
Evaluation ID: 7e462797-e01d-2a30-109f-1c9f76eef6fa

$ nomad job status example
...
Status        = running
...
Allocations
ID        Node ID   Task Group  Version  Desired  Status    Created    Modified
30229450  4786abea  cache       1        run      running   5s ago     4s ago
204c970f  4786abea  cache       0        stop     complete  32s ago    12s ago

$ nomad job stop example
Are you sure you want to stop multi-region job "example" in a single region? [y/N] y
==> Monitoring evaluation "22ff5e86"
    Evaluation triggered by job "example"
    Evaluation within deployment: "9280c9e0"
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "22ff5e86" finished with status "complete"
````

### CODE REVIEW

File: command/job_stop.go [hunk 1]
```diff
@@ -138,29 +144,62 @@ func (c *JobStopCommand) Run(args []string) int {
...
+	if job.IsMultiregion() && !global {
+		question := fmt.Sprintf(
+			"Are you sure you want to stop multi-region job %q in a single region? [y/N]", *job.ID)
+		code, confirmed := getConfirmation(question)
+		if !confirmed {
+			return code
+		}
+	}
+
+	// scatter-gather job stop for multi-region jobs
```
Comment (drewbailey) [inline command/job_stop.go:188]: nit
```suggestion
	// Scatter-gather job stop for multi-region jobs
```

File: command/job_stop.go [hunk 2]
```diff
@@ -138,29 +144,62 @@ func (c *JobStopCommand) Run(args []string) int {
...
+				return 1
+			}
+			c.Ui.Output(evalID)
+		}
+		return 0
+	}
+
 	// Invoke the stop
 	wq := &api.WriteOptions{Namespace: jobs[0].JobSummary.Namespace}
 	evalID, _, err := client.Jobs().Deregister(*job.ID, purge, wq)
```
Comment (drewbailey) [inline command/job_stop.go:205]: is this still necessary?

Comment (tgross) [reply command/job_stop.go:205]: That's where single-region jobs get stopped.

Comment (drewbailey) [reply command/job_stop.go:205]: ah sorry, didn't see return 0 in the multiregion block :(

**Pull request merged**